app/src/main/java/com/portfolio/tasky/entry/repositories/EntryRepository.kt

```diff
@@ -0,0 +1,70 @@
+package com.portfolio.tasky.entry.repositories
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.portfolio.tasky.entry.network.EntryApiCall
+import com.portfolio.tasky.entry.models.AuthenticationRequest
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import com.portfolio.tasky.entry.models.RegisterRequest
+import com.portfolio.tasky.networking.usecases.domain.TaskyCallStatus
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
+
+class EntryRepository @Inject constructor(private val entryApiCall: EntryApiCall)  {
+    private val registrationLiveData = MutableLiveData<Boolean?>()
+    private val loginLiveData = MutableLiveData<AuthenticationResponse?>()
+
+    fun doRegistration(taskyCallStatusCallback: TaskyCallStatus, registerModel: RegisterRequest) : LiveData<Boolean?>{
+
+        taskyCallStatusCallback.onRequestCallStarted()
+
+        entryApiCall.registration(registerModel).enqueue(object : Callback<Void>{
+            override fun onResponse(call: Call<Void>, response: Response<Void>) {
+                if(response.code() == 200){
+                    taskyCallStatusCallback.onResponse(response.code(), response.body().toString())
+                    registrationLiveData.postValue(true)
+                }else{
+                    val jObjError = response.errorBody()?.string()?.let { JSONObject(it) }
+                    taskyCallStatusCallback.onResponse(response.code(), jObjError?.getString("message") as String)
+                    registrationLiveData.postValue(false)
+                }
+
+            }
+
+            override fun onFailure(call: Call<Void>, t: Throwable) {
+                taskyCallStatusCallback.onFailure(0, t.message.toString())
+                registrationLiveData.postValue(false)
+            }
+
+        })
```

Comment:
If you remove Call + use coroutines, you can simply replace this with

```
try {
val result = entryApiCall.registration(registerModel)
// Do something with result
} catch(e: CancellationException) {
    throw e
} catch(e: Exception) {
// Do something here
}

```

Reply:
It will require some more code in order to make it work with current implementation

Reply:
Awesome!